### PR TITLE
EN-1441 Restore database backup endpoint

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/web/database/DatabaseController.java [hunk 1]
```diff
@@ -92,6 +92,17 @@
...
     }
 
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/restoreBackup/{reportCode}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> restoreBackup(@PathVariable String reportCode) throws Throwable {
+        logger.debug("Starting database restore");
+        this.getDatabaseService().startDatabaseRestore(reportCode);
+        Map<String, String> response = new HashMap<>();
+        response.put("status", String.valueOf(this.getDatabaseService().getStatus()));
+        logger.debug("Database restore started");
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/database/DatabaseController.java]: This should probably say completed. We should log the report code

File: engine/src/main/java/org/entando/entando/web/database/DatabaseController.java [hunk 2]
```diff
@@ -92,6 +92,17 @@
         return new ResponseEntity<>(new RestResponse(response), HttpStatus.OK);
     }
 
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/restoreBackup/{reportCode}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> restoreBackup(@PathVariable String reportCode) throws Throwable {
+        logger.debug("Starting database restore");
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/database/DatabaseController.java]: we should log the report code.

File: engine/src/main/java/org/entando/entando/web/database/DatabaseController.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/database/DatabaseController.java]: Done!

File: engine/src/main/java/org/entando/entando/web/database/DatabaseController.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/database/DatabaseController.java]: Done!

**Comment by entando-service** [comment]
All is well

**Pull request merged**